hw/dv/tools/ralgen/ralgen.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""FuseSoc generator for UVM RAL package created with either regtool or
+topgen tools.
+"""
+import argparse
+import os
+import sys
+from collections import OrderedDict
+
+import yaml
+
+try:
+    from yaml import CSafeLoader as YamlLoader, CSafeDumper as YamlDumper
+except ImportError:
+    from yaml import SafeLoader as YamlLoader, SafeDumper as YamlDumper
+
+
+def main():
+    if len(sys.argv) != 2:
+        print("ERROR: This script takes a single YAML file as input argument")
+        sys.exit(1)
+
+    gapi_filepath = sys.argv[1]
+    gapi = yaml.load(open(gapi_filepath), Loader=YamlLoader)
+
+    # This is just a wrapper around the reggen and topgen tools, which
+    # are referenced from proj_root area.
+    self_path = os.path.dirname(os.path.realpath(__file__))
+    util_path = os.path.abspath(os.path.join(self_path, "../../../../util"))
```

Comment:
This is quite fragile. It's probably OK for now, but please make it a constant at the top of the script, e.g. `REPO_TOP_RELPATH = '../../../..'`, and then use that here.